sencha-workspace/SlateTasksTeacher/app/controller/Dashboard.js

```diff
@@ -1,43 +1,457 @@
+
+        if (btn.dateSelected) {
+            me.assignStudentTaskRevision(btn.dateSelected);
+        } else {
+            datepicker = Ext.widget({
+                xtype: 'datepicker',
+                floating: true,
+                handler: function(picker, date) {
+                    picker.destroy();
+                    btn.dateSelected = date;
+                    return btn.setText('Re-Assign revision due on '+Ext.Date.format(date, 'm/d/y'));
+                }
+            });
+            datepicker.show().showAt(coords[0], coords[1] - datepicker.getHeight()); // subtract datepicker height from y coord to show above button.
+        }
+    },
+
+    onUnassignStudentTaskClick: function() {
+        var me = this,
+            taskRater = me.getTaskRater(),
+            studentTask = taskRater.getStudentTask();
+
+        Ext.Msg.confirm('Are you sure?', 'Do you want to unassign this student task? This can not be undone.', function(ans) {
+            if (ans === 'yes') {
+                me.unAssignStudentTask(studentTask);
+            }
+        });
+
+    },
+
+    onEditStudentTaskClick: function() {
+        var me = this,
+            taskRater = me.getTaskRater(),
+            studentTask = taskRater.getStudentTask();
+
+        taskRater.close();
+        me.editStudentTask(studentTask);
+    },
+
+    onStudentsStoreLoad: function() {
+        window.console.log('onstudentstoreload', arguments);
+    },
+
+    onAssigneeComboRender: function(combo) {
+        var me = this,
+            comboStore = combo.getStore(),
+            studentsStore = me.getStudentsStore();
+
+        comboStore.removeAll();
+        comboStore.add(studentsStore.getRange());
+        combo.setValueOnData();
+        // console.log('students updated');
+    },
+
+    onTasksGridCellClick: function(grid, taskId, studentId) {
+        var me = this,
+            dataStore = grid.getDataStore(),
+            studentTask = dataStore.getAt(dataStore.findBy(function(r) {
+                return r.get('TaskID') == taskId && r.get('StudentID') == studentId;
+            })),
+            taskStatus;
+
+        if (studentTask) {
+            taskStatus = studentTask.get('TaskStatus');
+            if (taskStatus === 'assigned') {
+                return me.editStudentTask(studentTask);
+            } else if (taskStatus === 'submitted' || taskStatus === 're-submitted' || taskStatus === 're-assigned') {
+                return me.rateStudentTask(studentTask);
+            } else if (taskStatus === 'completed') {
+                return me.rateStudentTask(studentTask, true);
+            }
+        }
+
+        return me.assignStudentTask(taskId, studentId);
+    },
+
+    onRateSkillClick: function(ratingView, ratingObject) {
+        var me = this,
+            taskRater = me.getTaskRater(),
+            studentTask = taskRater.getStudentTask();
+
+        Slate.API.request({
+            url: '/cbl/student-tasks/'+studentTask.getId()+'/rate',
```

Comment:
@nbey the `Ext.data.Model` class has a standard `toUrl()` method you can override and implement. Centralize this logic on the model so the code here can just be: `url: studentTask.toUrl() + '/rate'`